src/lib.rs

```diff
@@ -206,16 +226,17 @@ where
 
     /// Convenience function for `STATUS_REG` to confirm all three X, Y and
     /// Z-axis have new data available for reading by accel_raw and associated
-    /// function calls
+    /// function calls.
     pub fn is_data_ready(&mut self) -> Result<bool, Error<E>> {
         let value = self.get_status()?;
+
         Ok(value.zyxda)
     }
 
     /// Temperature sensor enable.
-    /// `TEMP_CGF_REG`: `TEMP_EN`, the BDU bit in `CTRL_REG4` is also set
+    /// `TEMP_CGF_REG`: `TEMP_EN`, the BDU bit in `CTRL_REG4` is also set.
     pub fn enable_temp(&mut self, enable: bool) -> Result<(), Error<E>> {
-        self.register_xset_bits(Register::TEMP_CFG, TEMP_EN, enable)?;
+        self.register_xset_bits(Register::TEMP_CFG, ADC_EN & TEMP_EN, enable)?;
```

Comment:
I didn't mean to commit this but it slipped through; since the ADC __must__ be enabled in order to use the temperature sensor, and we're writing the register anyways, does it make sense to set `ADC_EN` in `enable_temp` as well, just to be sure? I can remove this otherwise.

Reply:
I did notice because someday Id like to go through the rust stack for power usage, but figured today was not that day. Seems like with current API we have to enable it, because if its not enabled you have to remove access to your get_temp calls. One option is an into_temp that enables the reg and then hands them a lis3dh_temp device that has a trait with those calls available.

Reply:
It is documented that the ADC must be enabled in order to read the temperature sensor. I think for the time being it's probably best to just go the path of least resistance and leave it how it is.